ocs_ci/ocs/platform_nodes.py

```diff
@@ -1816,3 +1829,116 @@ def attach_nodes_to_cluster(self, node_list):
+
+    """
+    def __init__(self, node_conf, node_type, worker_count):
+        """
+        Initialize necessary variables
+
+        Args:
+            node_conf (dict): of node configuration
+            node_type (str): type of node to be created RHCOS/RHEL
+            worker_count (int): number of nodes to add to existing cluster
+
+        """
+        super(BareMetalUPINode, self).__init__()
+        self.aws = aws.AWS()
+        self.node_conf = node_conf
+        self.node_type = node_type
+        self.worker_count = worker_count
+        self.nodes_for_expansion = list()
+        self.current_worker_count = len(get_typed_nodes())
+        self.target_worker_count = (
+            self.current_worker_count + self.worker_count
+        )
+        if self.node_type.upper() == 'RHEL':
+            raise UnsupportedPlatformError(
+                f"Platform {self.node_type.upper()} is not supported"
+            )
+
+        if constants.BM_MAX_WORKER_COUNT < self.target_worker_count:
+            raise InsufficientResourceException(
+                "No Sufficient BM present",
+                f"Current Worker count:- {self.current_worker_count}",
+                f"Overall Worker count:- {self.target_worker_count}",
+                f"Total Allowed Worker count:- {constants.BM_MAX_WORKER_COUNT}"
+            )
+
+        nodes_obj = get_typed_nodes()
+        nodes_name = list()
+        for node in nodes_obj:
+            nodes_name.append(node.name)
+        self.mgmt_details = load_auth_config()['ipmi']
+
+        for machine in self.mgmt_details:
+            if self.mgmt_details[machine].get('extra_node'):
+                if machine not in nodes_name:
+                    self.nodes_for_expansion.append(machine)
+
+    def add_node(self):
+        """
+        Add nodes to the current cluster
+        """
+        logger.info(f"Adding Nodes of type {self.node_type}")
+        logger.info(
+            f"Existing worker nodes: {self.current_worker_count}, "
+            f"New nodes to add: {self.worker_count}"
+        )
+        self._pxe_boot_nodes(self.nodes_for_expansion[:self.worker_count])
+        # Adding Ips to DNS record
+        cluster_name = f"{constants.BM_DEFAULT_CLUSTER_NAME}.{config.ENV_DATA['base_domain']}"
+        zone_id = self.aws.get_hosted_zone_id(cluster_name=cluster_name)
+        for node in self.nodes_for_expansion:
+            logger.info(f"Updating *.apps DNS record with {self.mgmt_details[node]['ip']}")
+            self.aws.update_hosted_zone_record(
+                zone_id=zone_id, record_name=f'*.apps.{cluster_name}',
+                data=self.mgmt_details[node]['ip'], type='A', operation_type='Add'
+            )
+        # Gets the existing CSR data
+        existing_csr_data = get_nodes_csr()
+        pre_count_csr = len(existing_csr_data)
+        logger.debug(f"Existing CSR count before adding nodes: {pre_count_csr}")
+        if constants.CSR_BOOTSTRAPPER_NODE in existing_csr_data:
```

Comment:
It will be more clear if this "if" statement will be explained with a comment. I guess that we add one more or not because the **CSR_BOOTSTRAPPER_NODE** is not counted in the function `def get_nodes_csr`.